### PR TITLE
sysdeps/linux: implement a few more sysdeps

### PR DESCRIPTION
**Pull request by 64**
This PR is part of the mlibc on Linux project.

(Dennis forced me to write that)

### CODE REVIEW

**Comment by Dennisbonke** [comment]
> This PR is part of the mlibc on Linux project.
> 
> (Dennis forced me to write that)

I sure didn't, but it's true :^)

File: options/glibc/generic/glibc-signal.cpp [hunk 1]
```diff
@@ -1,7 +1,18 @@
+#include <errno.h>
 #include <signal.h>
 #include <bits/ensure.h>
+#include <mlibc/debug.hpp>
+#include <mlibc/posix-sysdeps.hpp>
```
Comment (Geertiebear) [inline options/glibc/generic/glibc-signal.cpp:5]: We have a ``glibc-sysdeps.hpp``

**Review by Geertiebear — CHANGES_REQUESTED** [review]
LGTM except for one comment.

File: sysdeps/linux/generic/sysdeps.cpp [hunk 2]
```diff
@@ -616,4 +771,69 @@ int sys_setrlimit(int resource, const struct rlimit *limit) {
...
+}
+
+int sys_dup(int fd, int flags, int *newfd) {
+	auto ret = do_cp_syscall(NR_dup, fd);
+	if (int e = sc_error(ret); e)
+		return e;
+	// TODO: Handle flags
+	(void) flags;
+	*newfd = sc_int_result<int>(ret);
+	return 0;
```
Comment (ArsenArsen) [inline sysdeps/linux/generic/sysdeps.cpp:810]: We can just call sys_dup (or NR_dup3) here

**Review by Geertiebear — APPROVED** [review]
LGTM.

**Pull request merged**